docs/source/common_issues.rst

```diff
@@ -821,3 +821,59 @@ This is best understood via an example:
 To get this code to type check, you could assign ``y = x`` after ``x`` has been
 narrowed, and use ``y`` in the inner function, or add an assert in the inner
 function.
+
+
+Upgrading to :pep:`585` syntax
+------------------------------
```

Comment:
This is slightly more verbose, but I think it's good to have "name shadowing" in the section title, since that's the root cause of the issue:

```suggestion
Name shadowing issues with :pep:`585` syntax
--------------------------------------------
```